Source/Spark.swift

```diff
@@ -24,31 +24,126 @@ import UIKit
 
 
 /// The main entry point into the SDK. Contains methods for initializing the SDK and verifying developer and end user credentials.
-open class Spark {
+public class Spark {
     
     /// The version number of this SDK.
-    open static let version = "1.1.0"
+    public static let version = "1.1.0"
     
-    /// Indicates whether the SDK has been authorized.
-    open static func authorized() -> Bool {
-        return AuthManager.sharedInstance.authorized()
+    /// Toggle to enable or disable console log output.
+    ///
+    /// - parameter enable: Set True to enable console log, False as not.
+    /// - returns: Void
+    public static func toggleConsoleLogger(_ enable: Bool) {
+        LoggerManager.sharedInstance.toggleConsoleLogger(enable)
+    }
+    
+    /// AuthenticationStrategy allows your application to check and modify authentication state
+    public let authenticationStrategy: AuthenticationStrategy
+    
+    /// Phone allows your application to make media calls on Spark.
+    public let phone: Phone
+    
+    /// CallNotificationCenter allows your application to be notified of call events
+    public let callNotificationCenter: CallNotificationCenter
+    
+    public init(authenticationStrategy: AuthenticationStrategy) {
+        self.authenticationStrategy = authenticationStrategy
+        let deviceService = DeviceService(authenticationStrategy: authenticationStrategy)
+        let callManager = CallManager(authenticationStrategy: authenticationStrategy, deviceService: deviceService)
+        let webSocketService = WebSocketService(authenticationStrategy: authenticationStrategy, callManager: callManager)
+        let applicationLifecycleObserver = ApplicationLifecycleObserver(webSocketService: webSocketService, callManager: callManager, deviceService: deviceService)
+        phone = Phone(authenticationStrategy: authenticationStrategy, applicationLifecycleObserver: applicationLifecycleObserver, webSocketService: webSocketService, callManager: callManager, deviceService: deviceService)
```

Comment:
Yeah, there's a lot I'd like to see done with that to be honest, we haven't got there yet. I'd like to see much cleaner state management, with some of those objects only existing after registration and living only as long as the registration.